From 1374e88c7dc9ad1199ea5a9cd00213ad55282ea9 Mon Sep 17 00:00:00 2001
From: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Date: Mon, 28 Nov 2022 10:40:18 +0200
Subject: [PATCH 46/63] gpio: s32cc: Avoid signal glitch on set_direction

Avoid signal glitches on set_direction callback by
setting the value before the direction of the signal
in case of output direction.

Issue: ALB-9425
Upstream-Status: Pending 

Signed-off-by: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 drivers/gpio/gpio-siul2-s32cc.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/gpio/gpio-siul2-s32cc.c b/drivers/gpio/gpio-siul2-s32cc.c
index 3ae28fb89e38..f38bb6c039ee 100644
--- a/drivers/gpio/gpio-siul2-s32cc.c
+++ b/drivers/gpio/gpio-siul2-s32cc.c
@@ -332,12 +332,13 @@ static int siul2_gpio_dir_out(struct gpio_chip *chip, unsigned int gpio,
 	int ret = 0;
 	struct siul2_gpio_dev *gpio_dev;
 
+	gpio_dev = to_siul2_gpio_dev(chip);
+	siul2_gpio_set_val(chip, gpio, val);
+
 	ret = pinctrl_gpio_direction_output(siul2_gpio_to_pin(chip, gpio));
 	if (ret)
 		return ret;
 
-	gpio_dev = to_siul2_gpio_dev(chip);
-	siul2_gpio_set_val(chip, gpio, val);
 	gpio_set_direction(gpio_dev, gpio, OUT);
 
 	return ret;
-- 
2.25.1

